internal/files/preallocate_linux.go

```diff
@@ -0,0 +1,16 @@
+//go:build linux
+
+package files
+
+import (
+	"os"
+
+	"golang.org/x/sys/unix"
+)
+
+func PreAllocate(file *os.File, size int64) error {
+	if size > 0 {
```

Comment:
The fuzz testing in CI caught this, the syscall does not allow you to pass a size of 0.